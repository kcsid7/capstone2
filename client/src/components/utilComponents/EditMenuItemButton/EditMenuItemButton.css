.EditMenuItemButton {
    margin: 0 0.5rem 0.7rem;
    width: 100px;
    padding: 0.6rem;
    font-size: 1rem;
    border: none;
    box-shadow: 5px 5px 5px rgba(0, 0, 0, 0.700);
    border-radius: 0.5rem;
}

.Save-Button {
    background-color: rgb(2, 154, 2);
    color: white;
    box-shadow: 5px 5px 5px green;

}

.Delete-Button {
    background-color: rgba(209, 26, 26);
    color: white;
    box-shadow: 5px 5px 5px red;
}