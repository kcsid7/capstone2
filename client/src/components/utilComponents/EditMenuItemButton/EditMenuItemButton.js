import React, {useState} from "react";

//MUI

import "./EditMenuItemButton.css"

function EditMenuItemButton({toggle, edit, deleteItem, updateItem}) {

    const editOffHTML = (
        <button 
            className="EditMenuItemButton" 
            onClick={toggle}>
                Edit Item
        </button>
    )
    
    const editOnHTML = (
        <div>
            <button 
                className="EditMenuItemButton Save-Button" 
                onClick={updateItem}>
                    Save
            </button>

            <button 
                className="EditMenuItemButton" 
                onClick={toggle}>
                    Cancel
            </button>

            <button 
                className="EditMenuItemButton Delete-Button" 
                onClick={deleteItem}>
                    Delete
            </button>
        </div>
    )

    return(
            edit ? 
            editOnHTML:
            editOffHTML
    )
}

export default EditMenuItemButton;