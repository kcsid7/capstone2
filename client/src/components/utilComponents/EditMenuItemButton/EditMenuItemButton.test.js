import React from "react";
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom'; 

import EditMenuItemButton from "./EditMenuItemButton";


// Smoke Test
test("Renders without crashing", () => {
    render(<BrowserRouter><EditMenuItemButton /></BrowserRouter>)
})


test("Count Button snapshot", () => {
    const { asFragment } = render(<BrowserRouter><EditMenuItemButton /></BrowserRouter>);
    expect(asFragment()).toMatchSnapshot();
})