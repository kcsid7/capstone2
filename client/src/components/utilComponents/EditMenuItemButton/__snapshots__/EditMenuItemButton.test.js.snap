// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Count Button snapshot 1`] = `
<DocumentFragment>
  <button
    class="EditMenuItemButton"
  >
    Edit Item
  </button>
</DocumentFragment>
`;
